sensor/common/registry/registry_store.go

```diff
@@ -474,14 +479,19 @@ func (rs *Store) upsertSecretByName(namespace, secretName string, dockerConfig c
 	rs.storeMutux.Lock()
 	defer rs.storeMutux.Unlock()
 
-	for registryAddress, dce := range dockerConfig {
-		registryAddr := strings.TrimSpace(registryAddress)
+	for registryRecord, dce := range dockerConfig {
+		// Remove http/https prefixes and path from registry, matching may fail otherwise.
+		// The created registry.url will have the appropriate prefix.
+		registry := strings.TrimSpace(registryRecord)
+		endpoint := urlfmt.GetServerFromURL(
+			urlfmt.FormatURL(registry, urlfmt.HTTPS, urlfmt.NoTrailingSlash),
+		)
 
 		if hasBoundServiceAccount {
 			// Registries found in any `dockercfg` secret bound a service account
 			// are assumed to reference the OCP internal registry.
-			rs.upsertPullSecretByNameNoLock(namespace, secretName, registryAddr, dce)
-			rs.addClusterLocalRegistryHost(registryAddr)
+			rs.upsertPullSecretByNameNoLock(namespace, secretName, registry, endpoint, dce)
+			rs.addClusterLocalRegistryHost(endpoint)
```

Comment:
I think it might be worth introducing a `registry` type:
```go
type Registry string
func (r Registry) getHostPort() string;
...
```
It is often not clear from the code what various variables are expected to contain:
```go
trimmed := urlfmt.TrimHTTPPrefixes(host)
```

Reply:
nit: FWIW, the current codebase has 3 related image registry types that are an existing source of confusion. 

[`Registry` - pkg/registries/types/types.go](https://github.com/stackrox/stackrox/blob/f6627c14c7fd6b38213b506d688ef3efaea88a34/pkg/registries/types/types.go#L70)
[`ImageRegistry` - pkg/registries/types/types.go](https://github.com/stackrox/stackrox/blob/f6627c14c7fd6b38213b506d688ef3efaea88a34/pkg/registries/types/types.go#L83)
[`Registry` - pkg/registries/docker/docker.go](https://github.com/stackrox/stackrox/blob/f6627c14c7fd6b38213b506d688ef3efaea88a34/pkg/registries/docker/docker.go#L56)

Adding another may create more confusion IMHO (please consider).

Reply:
decided not to add a fourth one 😅